diag_manager/fms_diag_time_reduction.F90

```diff
@@ -0,0 +1,218 @@
+!!
+!> @addtogroup fms_diag_time_reduction_mod
+   TYPE time_reduction_type
+      integer , private :: the_type !< The time reduction type; integer as per diag_data_mod entries.
+      logical , private :: time_averaging !< Set true iff time_average, time_rms, time_power or time_diurnal is true
+      logical , private :: time_ops !< Set true iff time_min, time_max, time_rms or time_average is true.
+   CONTAINS
+      procedure, public :: do_time_averaging => do_time_averaging_imp
+      procedure, public :: has_time_ops => has_time_ops_imp
+      procedure, public :: is_time_none => is_time_none_imp
+      procedure, public :: is_time_average => is_time_average_imp
+      procedure, public :: is_time_rms => is_time_rms_imp
+      procedure, public :: is_time_max => is_time_max_imp
+      procedure, public :: is_time_min => is_time_min_imp
+      procedure, public :: is_time_sum => is_time_sum_imp
+      procedure, public :: is_time_diurnal => is_time_diurnal_imp
+      procedure, public :: is_time_power => is_time_power_imp
+      procedure, public :: initialize
+   END TYPE time_reduction_type
+
+!> @brief This interface is for the class constructor.
+!> @addtogroup fms_diag_time_reduction_mod
+   interface  time_reduction_type
+      procedure  :: time_reduction_type_constructor
+   end interface time_reduction_type
+
+CONTAINS
+
+   !> @brief The class contructors. Just allocates the class and calls an initializer
+   function time_reduction_type_constructor(dt, out_frequency) result(time_redux)
+      integer, intent(in) :: dt  !> The redution type (time_rms, time_power, etc)
+      integer, intent(in) :: out_frequency  !> The output frequency.
+      class (time_reduction_type), allocatable :: time_redux
+      allocate(time_redux)
+      call time_redux%initialize(dt, out_frequency)
+   end function time_reduction_type_constructor
+
+!> @brief Initialize the object.
+   subroutine initialize(this, dt, out_frequency)
+      class (time_reduction_type), intent(inout) :: this !> The time_reduction_type object
+      integer, intent(in) :: dt !> The redution type (time_rms, time_porer, etc)
+      integer, intent(in) :: out_frequency !> The output frequency.
+
+      this%the_type = dt
+
+      !! Set the time_averaging flag
+      !! See legacy init_ouput_fields function, lines 1470ff
+      IF(( dt .EQ. time_average) .OR. (dt .EQ. time_rms) .OR. (dt .EQ. time_power) .OR. &
+      & (dt .EQ. time_diurnal)) THEN
+         this%time_averaging = .true.
+      ELSE
+         this%time_averaging= .false.
+         IF((dt .NE. time_max) .AND. (dt .ne. time_min) .AND. (dt .NE. time_sum) &
+         & .AND. (dt .NE. time_none)) THEN
+            CALL error_mesg('time_reduction_type: initialize', &
+            & 'time_averaging=.false. but reduction type not compatible', FATAL)
+         ENDIF
+      END IF
+
+      !!TODO: Add other checks? E.g. If time_averaging == .false., then
+      !!  out_frequency == EVERY_TIME
```

Comment:
Are you adding these checks?

Reply:
I wasn't planning to do it now, but its possible. Will have to talk to Uriel.